test/univariate/discrete/geometric.jl

```diff
@@ -0,0 +1,11 @@
+using Distributions
+using Test
+using ForwardDiff
+
+@testset "Geometric mgf and k vs k-1 parametrization #1604" begin
+    d = Geometric(0.2)
+    @test mgf(d, 0) ≈ 1 atol = 10eps(Float64) rtol=0
+    @test cf(d, 0)  ≈ 1 atol = 10eps(Float64) rtol=0
+    @test ForwardDiff.derivative(Base.Fix1(mgf, d), 0) ≈ mean(d)
+    @test ForwardDiff.derivative(Base.Fix1(cf, d), 0) ≈ mean(d) * im
+end
```

Comment:
Maybe we should check derivatives with FiniteDifferences instead since ForwardDiff sometimes silently returns wrong derivatives (https://github.com/JuliaStats/Distributions.jl/issues/1582) and finite differencing seems more reliable. It also avoids issues on old Julia versions mentioned below, I assume.

E.g., with the `expm1` changes above,
```suggestion
using FiniteDifferences

@testset "Geometric mgf and k vs k-1 parametrization #1604" begin
    d = Geometric(0.2)
    @test mgf(d, 0) == 1
    @test cf(d, 0) == 1

    fdm = central_fdm(5, 1)
    @test fdm(Base.Fix1(mgf, d), 0) ≈ mean(d)
    @test fdm(Base.Fix1(cf, d), 0) ≈ mean(d) * im
end
```